src/bindings/PyDP/algorithms/partition_selection_strategies.cpp

```diff
@@ -27,25 +24,74 @@ PyPartitionSelectionStrategy CreatePartitionStrategy(double epsilon, double delt
   return std::move(obj.value());
 }
 
-void init_algorithms_partition_selection_strategies(py::module& m) {
-  auto pyClass =
-      py::class_<dp::PartitionSelectionStrategy, PyPartitionSelectionStrategy>(
-          m, "PartitionSelectionStrategy",
-          R"pbdoc(
-        Base class for all (Ɛ, 𝛿)-differenially private partition selection strategies.
-      )pbdoc");
-  pyClass
-      .def("should_keep", &dp::PartitionSelectionStrategy::ShouldKeep,
-           py::arg("num_users"),
+template <class Strategy>
+py::class_<Strategy> init_partition_selection_strategy(py::module& m,
+                                                       const std::string& strategy_name,
+                                                       const std::string& docstring) {
+  auto py_class =
+      py::class_<Strategy>(m, strategy_name.c_str(), R"pbdoc(" + docstring + )pbdoc");
+  py_class
+      .def("should_keep", &Strategy::ShouldKeep, py::arg("num_users"),
            R"pbdoc(
               Decides whether or not to keep a partition with `num_users` based on differential privacy parameters and strategy.
             )pbdoc")
+      .def("probability_of_keep", &Strategy::ProbabilityOfKeep, py::arg("num_users"),
+           R"pbdoc(
+              Probability of keeping a partition with `num_users` based on differential privacy parameters and strategy.
+            )pbdoc")
+      .def_property_readonly("epsilon", &dp::PartitionSelectionStrategy::GetEpsilon)
+      .def_property_readonly("delta", &dp::PartitionSelectionStrategy::GetDelta)
+      .def_property_readonly(
+          "max_partitions_contributed",
+          &dp::PartitionSelectionStrategy::GetMaxPartitionsContributed)
       .attr("__module__") = "_partition_selection";
+  return py_class;
+}
+
+template <class Strategy>
+void add_thresholding_specific_methods(py::class_<Strategy>* py_class) {
+  py_class
+      ->def("noised_value_if_should_keep", &Strategy::NoiseValueIfShouldKeep,
+            py::arg("num_users"),
+            R"pbdoc(
+              Decides whether or not to keep a partition with `num_users` and
+              returns `num_users` + noise if the partition should be kept.
+            )pbdoc")
+      .def_property_readonly("threshold", &Strategy::GetThreshold);
+}
 
+void init_algorithms_partition_selection_strategies(py::module& m) {
+  // Truncated Geometric Partition selection strategy.
+  init_partition_selection_strategy<dp::PreaggPartitionSelection>(
+      m, "TruncatedGeometricPartitionSelectionStrategy",
+      "Truncated Geometric (epsilon, delta)-differenially private partition "
+      "selection strategy.");
   m.def("create_truncated_geometric_partition_strategy",
         &CreatePartitionStrategy<dp::PreaggPartitionSelection>);
+
+  // Laplace Partition selection strategy.
+  auto py_laplace_strategy_class =
```

Comment:
do we need auto here? Can we directly use the `Strategy` class?

Reply:
Yeah, it makes sense to use explicit type here. Done. PTAL